parts/site-content-heading.php

```diff
@@ -1,6 +1,88 @@
 <?php if ( is_search() ) : ?>
+
 	<div class="site-content-header">
-		<h1 class="site-content-header-title">Search results</h1>
-		<p>Showing results for <em>&ldquo;<?php the_search_query(); ?>&rdquo;</em></p>
+		<p class="site-content-header-title-pre">Search results</p>
+		<h1 class="site-content-header-title">&ldquo;<?php the_search_query(); ?>&rdquo;</h1>
 	</div>
+
+<?php elseif ( is_category() || is_tag() || is_tax() ) : ?>
+
+	<?php
+	$term  = get_queried_object();
+	$tax   = get_taxonomy( $term->taxonomy );
+	$title = single_term_title( '', false );
+	$desc  = term_description();
+	?>
+
+	<div class="site-content-header">
+
+		<p class="site-content-header-title-pre">
+			<?php printf( __( '%s Archive', 'handbook' ), esc_html( $tax->labels->singular_name ) ); ?>
+		</p>
+
+		<h1 class="site-content-header-title"><?php echo esc_html( $title ) ?></h1>
+
+		<?php if ( ! empty( $desc ) ) : ?>
+			<div class="site-content-header-title-description">
+				<?php echo wp_kses_post( $desc ); ?>
+			</div>
+		<?php endif; ?>
+
+	</div>
+
+<?php elseif ( is_author() ) : ?>
+
+	<?php
+	the_post(); // To grab the author from the first post.
```

Comment:
I don't think this is needed anymore, might be wrong, but I think that get fixed in Core.